src/test/java/de/wejago/hichi2influx/repository/InfluxDbRepositoryTest.java

```diff
@@ -16,46 +18,75 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.internal.util.reflection.Whitebox;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(MockitoExtension.class)
 class InfluxDbRepositoryTest {
-    private final String TIME_AS_STRING = "2022-02-16 02:25:32";
-
+   private final String TIME_AS_STRING = "2022-02-16 02:25:32";
     private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    private Queue<Point> measurementPoints;
 
     @Mock
     private InfluxDBClient influxDBClient;
-
     @Mock
-    private WriteApiBlocking writeApi;
-
+    private WriteApi writeApi;
     @Mock
     private InfluxDBConfig influxDBConfig;
-
     @InjectMocks
     private InfluxDbRepository influxDbRepository;
 
+    @BeforeEach
+    public void setup() {
+        influxDbRepository = new InfluxDbRepository(influxDBConfig);
+        measurementPoints = new LinkedBlockingQueue<>();
+    }
+
     @Test
-    void writePointShouldGenerateAndWriteMeasurementPoint() {
+    void shouldPersistPointsToInflux() {
         //GIVEN
-        SensorEntry sensorEntry = generateTestSensorEntry();
-        when(influxDBConfig.dbConnection()).thenReturn(influxDBClient);
-        when(influxDBClient.getWriteApiBlocking()).thenReturn(writeApi);
+        Point point = generateExpectedMeasurementPoint();
+        measurementPoints.add(point);
+        Whitebox.setInternalState(influxDbRepository, "measurementPoints", measurementPoints);
+        when(influxDBConfig.getWriteApi()).thenReturn(writeApi);
+
+        //WHEN
+        influxDbRepository.persistPointsToInflux();
+
+        //THEN
+        assertThat(measurementPoints).isEmpty();
+        verify(writeApi, times(1)).writePoint(point);
+    }
+
+    @Test
+    void shouldNotPersistPointsToInfluxWhenMeasurementPointsIsEmpty() {
+        influxDbRepository.persistPointsToInflux();
+
+        assertThat(measurementPoints).isEmpty();
+        verify(writeApi, never()).writePoint(any());
+    }
+
+    @Test
+    void shouldNotPersistPointsToInfluxWhenWriteApiIsNotAvailable() {
+        //GIVEN
+        Point point = generateExpectedMeasurementPoint();
+        measurementPoints.add(point);
+        Whitebox.setInternalState(influxDbRepository, "measurementPoints", measurementPoints);
+        when(influxDBConfig.getWriteApi()).thenReturn(null);
 
         //WHEN
-        Point actualPoint = influxDbRepository.writePoint(sensorEntry);
+        influxDbRepository.persistPointsToInflux();
 
         //THEN
-        Point expectedPoint = generateExpectedMeasurementPoint(sensorEntry);
-        verify(influxDBClient, times(1)).getWriteApiBlocking();
-        assertThat(actualPoint).usingRecursiveComparison()
-            .ignoringFields("time")
-            .isEqualTo(expectedPoint);
+        assertThat(measurementPoints).containsExactly(point);
```

Comment:
You don't need to check the content of the queue, verifying no write is enough in my opinion.